### ISSUE TITLE
Segmentation fault use mruby-cairo on FreeBSD/amd64

### ISSUE THREAD

**Issue by yamori813** [mruby/mruby#4397]
I try my mruby-cairo mrbgem on 2595d5f6c8b8d5f325b57bddb0683ddca7bba0e9. I have segmentation fault on FreeBSD/amd64.  I try on 2.0.0 source archive that is same.  But FreeBSD/arm and FreeBSD/mips have not problem on 2.0.0.

hiroki@sandbox:~/mruby % gdb ./build/host/bin/mruby 
GNU gdb 6.1.1 [FreeBSD]
Copyright 2004 Free Software Foundation, Inc.
GDB is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.
Type "show copying" to see the conditions.
There is absolutely no warranty for GDB.  Type "show warranty" for details.
This GDB was configured as "amd64-marcel-freebsd"...
(gdb) run /tmp/a
Starting program: /usr/home/hiroki/mruby/build/host/bin/mruby /tmp/a

Program received signal SIGSEGV, Segmentation fault.
0x000000000040ce9f in mrb_vm_exec (mrb=0x806426000, 
    proc=<value optimized out>, pc=0x8064f0776 ".\002\002\0037\002g")
    at /usr/home/hiroki/mruby/src/vm.c:1425
1425	        regs[bidx] = blk;
Current language:  auto; currently minimal
(gdb) where
#0  0x000000000040ce9f in mrb_vm_exec (mrb=0x806426000, 
    proc=<value optimized out>, pc=0x8064f0776 ".\002\002\0037\002g")
    at /usr/home/hiroki/mruby/src/vm.c:1425
#1  0x000000000040b75d in mrb_vm_run (mrb=0x806426000, proc=0x8064349f0, self=
      {value = {f = 1.7027901575617071e-313, p = 0x80643f9a0, i = 34464856480, sym = 105118112}, tt = MRB_TT_OBJECT}, stack_keep=0)
    at /usr/home/hiroki/mruby/src/vm.c:978
#2  0x0000000000415328 in mrb_top_run (mrb=<value optimized out>, 
    proc=0x8064349f0, self=
      {value = {f = 1.7027901575617071e-313, p = 0x80643f9a0, i = 34464856480, sym = 105118112}, tt = MRB_TT_OBJECT}, stack_keep=<value optimized out>)
    at value.h:223
#3  0x0000000000444abf in mrb_load_exec (mrb=0x806426000, 
    p=<value optimized out>, c=0x80650a090) at parse.y:6141
#4  0x0000000000403a61 in main (argc=<value optimized out>, 
    argv=<value optimized out>)
    at /usr/home/hiroki/mruby/mrbgems/mruby-bin-mruby/tools/mruby/mruby.c:280

hiroki@sandbox:~/mruby % lldb ./build/host/bin/mruby
(lldb) target create "./build/host/bin/mruby"
Current executable set to './build/host/bin/mruby' (x86_64).
(lldb) run /tmp/a
Process 32858 launching
Process 32858 launched: './build/host/bin/mruby' (x86_64)
Process 32858 stopped
* thread #1, name = 'mruby', stop reason = signal SIGSEGV: invalid address (fault address: 0x800000008)
    frame #0: 0x000000000040ce9f mruby`mrb_vm_exec(mrb=0x0000000806426000, proc=<unavailable>, pc=".\x02\x02\x037\x02g") at class.h:0
   1   	/*
   2   	** mruby/class.h - Class class
   3   	**
   4   	** See Copyright Notice in mruby.h
   5   	*/
   6   	
   7   	#ifndef MRUBY_CLASS_H
(lldb) bt
* thread #1, name = 'mruby', stop reason = signal SIGSEGV: invalid address (fault address: 0x800000008)
  * frame #0: 0x000000000040ce9f mruby`mrb_vm_exec(mrb=0x0000000806426000, proc=<unavailable>, pc=".\x02\x02\x037\x02g") at class.h:0
    frame #1: 0x000000000040b75d mruby`mrb_vm_run(mrb=0x0000000806426000, proc=0x00000008064349f0, self=mrb_value @ 0x0000000805820a40, stack_keep=0) at vm.c:978
    frame #2: 0x0000000000415328 mruby`mrb_top_run(mrb=<unavailable>, proc=0x00000008064349f0, self=mrb_value @ 0x0000000805820a90, stack_keep=<unavailable>) at vm.c:0
    frame #3: 0x0000000000444abf mruby`mrb_load_exec(mrb=0x0000000806426000, p=<unavailable>, c=0x000000080650a090) at parse.y:6141
    frame #4: 0x0000000000403a61 mruby`main(argc=<unavailable>, argv=<unavailable>) at mruby.c:280
    frame #5: 0x0000000000403215 mruby`_start + 149
(lldb) 

hiroki@sandbox:~/mruby % cat /tmp/a
c = Cairo.new(120, 160)
c.set_source_rgb(1, 0, 0)

**Comment by dearblue** [comment]
I did not reproduce it with gcc on FreeBSD 12.0/amd64, but I reproduced it when using the clang with optimization option.

From the conclusion, there is a problem with `mruby-cairo`.
It seems that the call stack is being destroyed by making `int` the place where` mrb_int` should be passed to `mrb_get_args()`.
After correction, segmentation fault no longer occurs.
Because it is a good opportunity, I will PR it to `mruby-cairo`.

When it became mruby-2.0, in the case of a 64-bit CPU, `mrb_int` is equivalent to `int64_t` because it is the same as `MRB_INT64` in default configuration.
Before that, it seems that the problem did not reveal because `mrb_int` was `int32_t` for both 64-bit and 32-bit CPUs in default configuration.
